client/extensions/woocommerce/app/settings/email/email-settings/components/internal-notification.js

```diff
@@ -13,49 +13,84 @@ import CompactFormToggle from 'components/forms/form-toggle/compact';
-		<ListItemField className="components__notification-component-title" >
-			{ ! isPlaceholder
-				? <FormLabel>
-					{ item.title }
-				</FormLabel>
-				: <p className="components__is-placeholder" />
+const InternalNotification = ( { item, recipient, checked, onChange, isPlaceholder, placeholder } ) => {
+	//Add field name to returned value
+	const toggle = ( value ) => {
+		onChange(
+			{
+				setting: item.field,
+				option: 'enabled',
+				value: value ? 'yes' : 'no',
 			}
-			{ ! isPlaceholder
-				? <FormSettingExplanation>
-					{ item.subtitle }
-				</FormSettingExplanation>
-				: <p className="components__is-placeholder" />
-			}
-		</ListItemField>
-		<ListItemField className="components__notification-component-input" >
-				<FormTextInput
-					className={ isPlaceholder ? 'components__is-placeholder' : null }
-					name={ item.field }
-					onChange={ onChange }
-					value={ recipient }
-				/>
-		</ListItemField>
-		<ListItemField className="components__notification-component-toggle">
-			{ ! isPlaceholder
-				? <CompactFormToggle
-					checked={ checked }
-					onChange={ onToggle }
-					id={ item.field }
-				/>
-				: <p className="components__is-placeholder" />
+		);
+	};
+
+	const change = ( { target: { value } } ) => {
+		onChange(
+			{
+				setting: item.field,
+				option: 'recipient',
+				value,
 			}
-		</ListItemField>
-	</ListItem>
-);
+		);
+	};
+
+	const checkedEmails = checkEmails( recipient );
+	const emailValidationError = checkedEmails.error;
+
+	return (
+		<ListItem className="components__notification-component-item">
+			<ListItemField className="components__notification-component-title" >
+				{ ! isPlaceholder
+					? <FormLabel>
+						{ item.title }
+					</FormLabel>
+					: <p className="components__is-placeholder" />
+				}
+				{ ! isPlaceholder
+					? <FormSettingExplanation>
+						{ item.subtitle }
+					</FormSettingExplanation>
+					: <p className="components__is-placeholder" />
+				}
+			</ListItemField>
+			<ListItemField className="components__notification-component-input" >
+					<FormTextInput
+						className={ isPlaceholder ? 'components__is-placeholder' : null }
+						isError={ emailValidationError }
+						name={ item.field }
+						onChange={ change }
+						value={ recipient }
+						placeholder={ placeholder }
+					/>
+					{ emailValidationError && (
+					<FormTextValidation
+						isError={ true }
+						text={ checkedEmails.messages[ 0 ].msg }
+					/>
+				) }
+			</ListItemField>
+			<ListItemField className="components__notification-component-toggle">
+				{ ! isPlaceholder
+					? <CompactFormToggle
+						checked={ checked }
+						onChange={ toggle }
+						id={ item.field }
+					/>
+					: <p className="components__is-placeholder" />
+				}
+			</ListItemField>
+		</ListItem>
+	);
+};
 
 InternalNotification.propTypes = {
 	checked: PropTypes.bool,
 	recipient: PropTypes.string,
 	item: PropTypes.object,
-	onToggle: PropTypes.func.isRequired,
```

Comment:
Yes but under different names because they have different purpose. isPlaceholder is for visual representation and placeholder is default value used when there is nothing in input field. I think that I should not use placeholder here but substitute value with the default when there is nothing originally in input. Good observation!